### PR TITLE
Add Override urgency to config

### PR DESCRIPTION
**Pull request by mildred**
Allow to override the urgency of some notifications from the config file

### CODE REVIEW

**Comment by ErikReider** [comment]
Thanks for the PR! Request a review from me when you're done with the draft :)

File: src/configModel/configModel.vala [hunk 1]
```diff
@@ -200,8 +200,47 @@ namespace SwayNotificationCenter {
         }
     }
 
+    public enum NotificationUrgencyEnum {
+        UNSET = -1,
+        LOW = 0,
+        NORMAL = 1,
+        CRITICAL = 2;
+
+        public static NotificationUrgencyEnum from_value (string value) {
```
Comment (mildred) [inline src/configModel/configModel.vala:209]: For some reason, I believe that this function is never called by the JSON deserialization process (I tried to run `value.ascii_down()` but it was not called). Same for the enum just above.

Comment (ErikReider) [reply src/configModel/configModel.vala:209]: The JSON parser checks this automatically. I created such methods for other enums for use in other methods

File: src/configSchema.json [hunk 2]
```diff
@@ -224,6 +224,12 @@
               "default": "Normal",
               "enum": ["Low", "Normal", "Critical"]
             },
+            "override-urgency": {
+              "type": "string",
+              "description": "The new urgency of the notification (optional)",
+              "default": "Unset",
```
Comment (ErikReider) [inline src/configSchema.json:230]: `Unset` should be lower-case just like below

File: src/configSchema.json [hunk 3]
```diff
@@ -224,6 +224,12 @@
...
+              "enum": ["unset", "low", "normal", "critical"]
+            },
```
Comment (ErikReider) [inline src/configSchema.json:232]: Make sure to update the Man(5) page

File: src/configModel/configModel.vala [hunk 4]
```diff
@@ -200,8 +200,47 @@ namespace SwayNotificationCenter {
         }
     }
 
+    public enum NotificationUrgencyEnum {
+        UNSET = -1,
+        LOW = 0,
+        NORMAL = 1,
+        CRITICAL = 2;
```
Comment (ErikReider) [inline src/configModel/configModel.vala:207]: Use the `UrgencyLevels` enums as values for these. Ex: `LOW = UrgencyLevels.LOW,`

File: src/configModel/configModel.vala [hunk 5]
```diff
@@ -200,8 +200,47 @@ namespace SwayNotificationCenter {
...
+                case "low":
+                    return LOW;
+                case "normal":
+                    return NORMAL;
+                case "critical":
+                    return CRITICAL;
+                default:
+                    return UNSET;
+            }
+        }
```
Comment (ErikReider) [inline src/configModel/configModel.vala:220]: Unused, should be removed

File: src/configModel/configModel.vala [hunk 6]
```diff
@@ -200,8 +200,47 @@ namespace SwayNotificationCenter {
...
+                case UNSET:
+                    return "Unset";
+            }
+            return "Unset";
+        }
+    }
+
     public class NotificationVisibility : NotificationMatching {
         public NotificationStatusEnum state { get; set; }
+        public NotificationUrgencyEnum override_urgency { get; set; default = NotificationUrgencyEnum.UNSET; }
```
Comment (ErikReider) [inline src/configModel/configModel.vala:243]: I'd prefer if this wasn't on a single line. Like this:
```vala
public NotificationUrgencyEnum override_urgency {
    get; set; default = NotificationUrgencyEnum.UNSET;
}
```

File: src/configModel/configModel.vala [hunk 7]
```diff
@@ -200,8 +200,47 @@ namespace SwayNotificationCenter {
...
+                case LOW:
+                    return "Low";
+                case NORMAL:
+                    return "Normal";
+                case CRITICAL:
+                    return "Critical";
+                case UNSET:
+                    return "Unset";
+            }
+            return "Unset";
```
Comment (ErikReider) [inline src/configModel/configModel.vala:237]: Join these with a `default`. Fewer duplicated strings :)

**Review by ErikReider — CHANGES_REQUESTED** [review]
Other than that, LGTM! :)

**Review by ErikReider — APPROVED** [review]
LGTM. Thanks for the help! :)

**Pull request merged**